### PR TITLE
[primTorch] Adding activation references for celu, mish, selu, softplus, and tanh

### PR DESCRIPTION
**Pull request by kevinstephano**
Adding activation references for celu, softplus, mish, selu.

### CODE REVIEW

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:45]: Nice workaround for div not being implemented yet -- we should add div

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:29]: .celu

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:49]: This is completely correct and consistent with the paper; note that our documentation for Celu is wrong for negative alpha

File: torch/testing/_internal/common_methods_invocations.py
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:18249]: This is perfect

**Comment by mruberry** [comment]
Looks brilliant so far, @kevinstephano; were it not for the odd lint issue we could merge it now!

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [reply torch/_refs/nn/functional/__init__.py:45]: Good news -- you can use refs.true_divide here, see

https://github.com/pytorch/pytorch/blob/2f602abf1454ccb21a012a8bbfe48f448c983467/torch/_refs/__init__.py#L702

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:25]: After rebasing this will be KIND.DEFAULT (because OP_MATH was renamed to DEFAULT)

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:101]: This is probably not what we want to do for alpha. If we want to verify the input is a floating dtype or higher then we should just check for that directly with utils.is_float_dtype and utils._is_complex_dtype. But we can probably avoid the type checking completely here

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:120]: Is this fixed after https://github.com/pytorch/pytorch/pull/77542?

File: torch/testing/_internal/common_methods_invocations.py
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:18271]: Nice toleranceOverride for CPU -- @ezyang also has a suggestion https://github.com/pytorch/pytorch/issues/77687 that would avoid us having to add these in the future

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:128]: I don't totally understand the current test failures, because they seem to be complaining about torch.nn.functional.softplus, and not refs.softplus - maybe we have a bug in our tests?

However, they are hinting at a legitimate issues, which is that softplus actually has a secret out= parameter:

https://github.com/pytorch/pytorch/blob/dcd2ba353844bbc7d480bf3aa15fd0eeb0d55e78/aten/src/ATen/native/native_functions.yaml#L9566

So this probably needs to apply the out_wrapper from _/prims/wrappers.py

File: torch/_refs/nn/functional/__init__.py
Comment (kevinstephano) [reply torch/_refs/nn/functional/__init__.py:120]: I am going to merge with #77542 and find out.  I was holding off from removing the `WIP` badge until I made the merge.

File: torch/_refs/nn/functional/__init__.py
Comment (kevinstephano) [reply torch/_refs/nn/functional/__init__.py:128]: I think this was my fault as I added `supports_out=False` to the `softplus` `opInfo` entry because I was getting the following error when testing `_refs.softplus`.  

```
======================================================================
ERROR: test_out__refs_nn_functional_softplus_cpu_float32 (__main__.TestCommonCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 377, in instantiated_test
    result = test(self, **param_kwargs)
  File "/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 786, in test_wrapper
    return test(*args, **kwargs)
  File "test/test_ops.py", line 738, in test_out
    _compare_out(_case_zero_transform)
  File "test/test_ops.py", line 713, in _compare_out
    op_out(out=out)
  File "/workspace/pytorch/torch/testing/_internal/common_methods_invocations.py", line 930, in __call__
    return self.op(*args, **kwargs)
  File "/workspace/pytorch/torch/_prims/wrappers.py", line 95, in _fn
    bound = sig.bind(*args, **kwargs)
  File "/opt/conda/lib/python3.8/inspect.py", line 3037, in bind
    return self._bind(args, kwargs)
  File "/opt/conda/lib/python3.8/inspect.py", line 3026, in _bind
    raise TypeError(
TypeError: got an unexpected keyword argument 'out'
```

**Comment by kevinstephano** [comment]
@mruberry This PR is ready to be reviewed.

File: torch/_refs/nn/functional/__init__.py
Comment (mruberry) [inline torch/_refs/nn/functional/__init__.py:117]: Nice

**Review by mruberry — APPROVED** [review]
Awesome!

Thanks @kevinstephano! Tomorrow let's talk about some nvFuser integration stuff

**Pull request closed without merging**
